python/equistore-operations/equistore/operations/_dispatch.py

```diff
@@ -97,13 +99,12 @@ def bincount(input, weights=None, minlength=0):
         raise TypeError(UNKNOWN_ARRAY_TYPE)
 
 
-def array_like_data(array, data):
-    """Return `data` with the same data-type and array-type of array"""
+def list_to_array(array, data: List):
+    """Create an object from data wit the same type as `array`."""
     if isinstance(array, TorchTensor):
-        _check_all_torch_tensor([data])
-        return torch.Tensor(data, device=array.device)
+        tensor = torch.Tensor(data, device=array.device)
```

Comment:
```suggestion
        tensor = torch.tensor(data, device=array.device, dtype=array.dtype)
```

this does not work?

Reply:
I tried this at some point and it failed with a weird error message. I tried it again 5 minutes ago and it worked... I change it. Thanks for the suggestion!

Reply:
`torch.Tensor` and `torch.tensor` are different things, which might explain why one works and not the other. The documentation seems to point people toward the function `torch.tensor`, but if the code works with the class constructor here, that's fine for me.